platform/api/brightbox/api.go

```diff
@@ -0,0 +1,234 @@
+		opts:   opts,
+	}, nil
+}
+
+func (a *API) AddKey(name, key string) error {
+	return nil
+}
+
+// CreateServer using the Brightbox API.
+func (a *API) CreateServer(name, userdata, cloudIP string) (*Server, error) {
+	// Not in the spec, but userdata needs to be base64 encoded.
+	userdata = base64.StdEncoding.EncodeToString([]byte(userdata))
+
+	s, err := a.client.CreateServer(context.TODO(), brightbox.ServerOptions{
+		Image:      &a.opts.Image,
+		Name:       &name,
+		UserData:   &userdata,
+		ServerType: &a.opts.ServerType,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("creating server from API: %w", err)
+	}
+
+	// If the cloud IP already exists, we reuse it - otherwise we create a new one.
+	if cloudIP == "" {
+		plog.Info("No cloud IP already available: creating a new one.")
+		cip, err := a.client.CreateCloudIP(context.TODO(), brightbox.CloudIPOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("creating a cloud IP from API: %w", err)
+		}
+
+		cloudIP = cip.ID
+	}
+
+	// Let's assign this IP to this new server.
+	if _, err := a.client.MapCloudIP(context.TODO(), cloudIP, brightbox.CloudIPAttachment{Destination: s.ID}); err != nil {
+		_ = a.DeleteServer(s.ID)
+		return nil, fmt.Errorf("mapping cloud IP to server: %w", err)
+	}
+
+	// Refetch the server to get the new information regarding the freshly assigned cloud IP.
+	s, err = a.client.Server(context.TODO(), s.ID)
+	if err != nil {
+		_ = a.DeleteServer(s.ID)
+		return nil, fmt.Errorf("getting server from API: %w", err)
+	}
+
+	return &Server{s}, nil
+}
+
+func (a *API) DeleteKey(name string) error {
+	return nil
+}
+
+// DeleteImage will remove the image from Brightbox.
+func (a *API) DeleteImage(id string) error {
+	if _, err := a.client.DestroyImage(context.TODO(), id); err != nil {
+		return fmt.Errorf("destroying image from API: %w", err)
+	}
+
+	return nil
+}
+
+// DeleteCloudIP will remove a cloud IP from Brightbox.
+func (a *API) DeleteCloudIP(id string) error {
+	if _, err := a.client.DestroyCloudIP(context.TODO(), id); err != nil {
+		return fmt.Errorf("destroying cloud IP from API: %w", err)
+	}
+
+	return nil
+}
+
+func (a *API) DeleteServer(id string) error {
+	// Let's first unassign the cloud IP.
+	s, err := a.client.Server(context.TODO(), id)
+	if err != nil {
+		return fmt.Errorf("getting server from API: %w", err)
+	}
+
+	var cloudIP string
+	if s != nil && len(s.CloudIPs) >= 1 {
+		cloudIP = s.CloudIPs[0].ID
+	}
+
+	if cloudIP != "" {
+		if _, err := a.client.UnMapCloudIP(context.TODO(), cloudIP); err != nil {
+			return fmt.Errorf("unmaping cloud IP from API: %w", err)
+		}
+		plog.Info("Cloud IP released.")
+	}
+
+	if _, err := a.client.DestroyServer(context.TODO(), id); err != nil {
+		return fmt.Errorf("destroying server from API: %w", err)
+	}
+
+	return nil
+}
+
+func (a *API) GC(gracePeriod time.Duration) error {
+	threshold := time.Now().Add(-gracePeriod)
+	// TODO: CloudIP has no creation date for now.
+	// We can't safely delete "old" cloud IPs.
```

Comment:
It would work to add an explicit removal of all cloud IPs as ore subcommand which we run before `kola run`.